src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -0,0 +1,7757 @@
+                    emit->emitIns_R_R_I(INS_sltiu, EA_PTRSIZE, targetReg, targetReg, 1);
+                }
+            }
+        }
+        else
+        {
+            regNumber regOp2 = op2->GetRegNum();
+
+            if (cmpSize == EA_4BYTE)
+            {
+                regNumber tmpRegOp1 = REG_RA;
+                regNumber tmpRegOp2 = rsGetRsvdReg();
+                assert(regOp1 != tmpRegOp2);
+                assert(regOp2 != tmpRegOp2);
+
+                if (IsUnsigned)
+                {
+                    emit->emitIns_R_R_I(INS_slli, EA_8BYTE, tmpRegOp1, regOp1, 32);
+                    emit->emitIns_R_R_I(INS_srli, EA_8BYTE, tmpRegOp1, tmpRegOp1, 32);
+
+                    emit->emitIns_R_R_I(INS_slli, EA_8BYTE, tmpRegOp2, regOp2, 32);
+                    emit->emitIns_R_R_I(INS_srli, EA_8BYTE, tmpRegOp2, tmpRegOp2, 32);
+                }
+                else
+                {
+                    emit->emitIns_R_R_I(INS_slliw, EA_8BYTE, tmpRegOp1, regOp1, 0);
+                    emit->emitIns_R_R_I(INS_slliw, EA_8BYTE, tmpRegOp2, regOp2, 0);
+                }
+
+                regOp1 = tmpRegOp1;
+                regOp2 = tmpRegOp2;
+            }
+
+            if (tree->OperIs(GT_LT))
+            {
+                emit->emitIns_R_R_R(IsUnsigned ? INS_sltu : INS_slt, EA_8BYTE, targetReg, regOp1, regOp2);
+            }
+            else if (tree->OperIs(GT_LE))
+            {
+                emit->emitIns_R_R_R(IsUnsigned ? INS_sltu : INS_slt, EA_8BYTE, targetReg, regOp2, regOp1);
+                emit->emitIns_R_R_I(INS_xori, EA_PTRSIZE, targetReg, targetReg, 1);
+            }
+            else if (tree->OperIs(GT_GT))
+            {
+                emit->emitIns_R_R_R(IsUnsigned ? INS_sltu : INS_slt, EA_8BYTE, targetReg, regOp2, regOp1);
+            }
+            else if (tree->OperIs(GT_GE))
+            {
+                emit->emitIns_R_R_R(IsUnsigned ? INS_sltu : INS_slt, EA_8BYTE, targetReg, regOp1, regOp2);
+                emit->emitIns_R_R_I(INS_xori, EA_PTRSIZE, targetReg, targetReg, 1);
+            }
+            else if (tree->OperIs(GT_NE))
+            {
+                emit->emitIns_R_R_R(INS_xor, EA_PTRSIZE, targetReg, regOp1, regOp2);
+                emit->emitIns_R_R_R(INS_sltu, EA_PTRSIZE, targetReg, REG_R0, targetReg);
+            }
+            else if (tree->OperIs(GT_EQ))
+            {
+                emit->emitIns_R_R_R(INS_xor, EA_PTRSIZE, targetReg, regOp1, regOp2);
+                emit->emitIns_R_R_I(INS_sltiu, EA_PTRSIZE, targetReg, targetReg, 1);
+            }
+        }
+    }
+}
+
+//------------------------------------------------------------------------
+// genCodeForJcc: Generate code for a GT_JCC node.
+//
+// Arguments:
+//    jcc - The node
+//
+void CodeGen::genCodeForJcc(GenTreeCC* jcc)
+{
+    emitter* emit = GetEmitter();
+
+    instruction ins = INS_invalid;
+
+    assert(jcc->OperIs(GT_JCC) && (instruction)jcc->GetRegNum() == INS_bnez);
```

Comment:
Why is `GetRegNum()` storing `instruction` here? Looks odd.

It's likely you should avoid producing `GT_JCC` at all for RISC-V. `GT_JCC` is based on using the CPU flags from the special flags register that exists on x64/x86/arm32/arm64. If this does not exist on `RISC-V` then you should just stick to `GT_JTRUE`/`GT_JCMP`.

I suggest that you introduce your own version of `LowerJTrue` inside lowerriscv64.cpp, similar to LA64. Then you can ensure that not `JCC` node is ever produced (this is the only place that creates `JCC`).